core/file_walker.go

```diff
@@ -4,17 +4,57 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"sync"
 )
 
 type File struct {
-	Name  string
-	Size  int64
-	IsDir bool
-	Files []*File
+	Name     string
+	Size     int64
+	IsDir    bool
+	Files    []*File
+	IsMarked bool
 }
 
 type ReadDir func(dirname string) ([]os.FileInfo, error)
 
+func (f *File) ToggleIsMarked() bool {
+	f.IsMarked = !f.IsMarked
+	return f.IsMarked
+}
+
+func GetMarked(root File) []string {
+	var marked []string
+	var wg sync.WaitGroup
+	c := make(chan string, 1)
+	wg.Add(1)
+	go func() {
+		wg.Wait()
+		close(c)
+	}()
+	go getMarkedRecursive("", &root, c, &wg)
+	for file := range c {
+		marked = append(marked, file)
+	}
+	return marked
+}
+
+func getMarkedRecursive(parent string, file *File, c chan<- string, wg *sync.WaitGroup) {
+	defer wg.Done()
+	path := "."
```

Comment:
@leonklingele Please have a look if you could use or somehow change `core.State.Path()` for marking. `Path()` was originally designed for status line on top of the screen. Ideally we could use it for both.